lib/mayaUsd/fileio/orphanedNodesManager.cpp

```diff
@@ -163,8 +169,22 @@ void renamePulledObject(
     for (const PullVariantInfo& info : trieNode->data()) {
         const MDagPath& mayaPath = info.editedAsMayaRoot;
         TF_VERIFY(writePullInformation(pulledPath, mayaPath));
-        if (usdPathChanged) {
-            TF_VERIFY(reparentPulledObject(mayaPath, pulledPath.pop()));
+        if (usdPathChanged && mayaPath.isValid()) {
+#ifdef MAYA_USD_AE_CRASH_ON_RENAME_WORKAROUND
+            // Update the proxyAccessor connections on idle to avoid a crash when
+            // attribute editor is visible and showing the ancestor USD prim beeing renamed.
+            if (MGlobal::mayaState() == MGlobal::kInteractive) {
```

Comment:
Is there a reason why we would only do this for interactive Maya? I know that sometimes when in non-interactive, on-idle are "too late" when scripting, is that the issue? It is somewhat unfortunate if it is, as it means that scripts run inside Maya will have the same problem. We do resort to this same kind of fix elsewhere, so it is not the first time, though.

Reply:
Yes, that’s exactly it; I wasn’t sure if idle would even be triggered in a non-interactive Maya session. It's a similar logic to mel's `evalDeferred`, which performs an immediate `eval` when in batch mode. Since the issue is related to the Attribute Editor, I chose to apply this workaround in interactive sessions only. It's not an ideal solution even for interactive, but unfortunately I haven't been able to find a better fix.